### PR TITLE
Add MembershipManagerInterface

### PR DESCRIPTION
**Pull request by amitaibu**
#296

### CODE REVIEW

**Comment by amitaibu** [comment]
Update: I have created the manager, and kept only the most used proxy methods in `\Drupal\og\Og`. Next is working on the tests.

**Comment by amitaibu** [comment]
Most tests fixed - we are now down to the last two unit tests methods

**Comment by amitaibu** [comment]
I believe tests should return green now..

**Comment by amitaibu** [comment]
Cool, tests are green. I'll just rename class to `MembershipManager` and it should be good to go.

**Comment by amitaibu** [comment]
Ready for review.

PR mostly moves methods to a new service, and keeps only a few methods in `\Drupal\og\Og`.
Also since we have this new service, the `::setup` for unit tests for `::userAccess` have been cleaned up a bit

File: src/OgAccess.php [hunk 1]
```diff
@@ -165,29 +175,20 @@ public function userAccess(EntityInterface $group, $operation, AccountInterface
...
+            $user_is_group_admin = TRUE;
+            break;
           }
+
+          $permissions = array_merge($permissions, $role->getPermissions());
         }
       }
-      else {
-        // User is a non-member.
+      elseif (!Og::isMemberBlocked($group, $user)) {
```
Comment (amitaibu) [inline src/OgAccess.php:190]: I've also cleaned up a bit this block of text, to reduce indentation.

**Comment by amitaibu** [comment]
I prefer to get things in, so it will unblock other PRs. We can always revisit later.

/cc @pfrenssen @RoySegall

**Comment by RoySegall** [comment]
Looks good to me except for this

``` php
/** @var \Drupal\og\MembershipManagerInterface $membership_manager */
$membership_manager = \Drupal::service('og.membership_manager');
```

This part appear multiple times and I think, IMO, we can convert it to:

``` php
$membership_manager = Og::membershipManager();
```

**Pull request merged**